### ISSUE TITLE
Git autostash doesn't work on 1.74.0

### ISSUE THREAD

**Issue by freewood** [microsoft/vscode#168861]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 
```
Version: 1.74.0 (user setup)
Commit: 5235c6bb189b60b01b1f49062f4ffa42384f8c91
Date: 2022-12-05T16:38:16.075Z
Electron: 19.1.8
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
```
- OS Version: Windows 10 21H2, VSCode server inside WSL Ubuntu 22.04 (same issue on the Debian 11)

Steps to Reproduce:

1. Enable autostash `"git.autoStash": true`
2. Have changes in the worktree
3. Press Sync
4. There you got window with git error
If we look to git output, we will see that `git stash pop` executed in the wrong order, before the pull
```
2022-12-12 15:52:01.762 [info] > git stash push -u [265ms]
2022-12-12 15:52:01.778 [info] > git stash pop [8ms]
2022-12-12 15:52:02.255 [info] > git pull --tags -r origin master [489ms]
```

**Comment by freewood** [comment]
Additional information. If we have staged changes in worktree and click "Commit & Sync", then order of commands is correct.
```
2022-12-13 13:12:56.053 [info] > git stash push -u [285ms]
2022-12-13 13:12:56.539 [info] > git pull --tags -r origin master [480ms]
2022-12-13 13:12:57.273 [info] > git push origin master:master [1206ms]
2022-12-13 13:12:57.381 [info] > git stash pop [102ms]
```

**Comment by freewood** [comment]
Fixed in 1.74.1

**Comment by alexr00** [comment]
Marking as verified by issue author!
